### PR TITLE
UI test stabilization

### PR DESCRIPTION
**Pull request by ouchadam**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [x] Technical
- [ ] Other :

## Content

- Always attempts to close the soft keyboard
- Avoids creating multiple rooms with the same name (empty room)

## Motivation and context

To get `./gradlew connectedGplayDebugAndroidTest -Pandroid.testInstrumentationRunnerArguments.class=im.vector.app.ui.UiAllScreensSanityTest` to consistently pass locally across different emulators

This check also caught #6511  

## Tests

No business logic changes

## Tested devices

- [ ] Physical
- [x] Emulator
- OS version(s): 28, 31v2

### CODE REVIEW

**Comment by ouchadam** [comment]
I haven't includef a changelog for this as it's only part of the test source set, happy to include one if wanted

File: vector/src/androidTest/java/im/vector/app/ui/robot/ElementRobot.kt [hunk 1]
```diff
@@ -91,8 +92,7 @@ class ElementRobot {
         waitUntilActivityVisible<CreateDirectRoomActivity> {
             waitUntilViewVisible(withId(R.id.userListSearch))
         }
-        // close keyboard
-        pressBack()
+        closeSoftKeyboard()
```
Comment (bmarty) [inline vector/src/androidTest/java/im/vector/app/ui/robot/ElementRobot.kt:95]: 💯

**Review by bmarty — APPROVED** [review]
LGTM, thanks!

File: vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt [hunk 2]
```diff
@@ -99,30 +99,33 @@ class UiAllScreensSanityTest {
...
         testThreadScreens()
 
+        val spaceName = UUID.randomUUID().toString()
         elementRobot.space {
             createSpace {
-                crawl()
+                createAndCrawl(spaceName)
             }
-            val spaceName = UUID.randomUUID().toString()
+            val publicSpace = UUID.randomUUID().toString()
```
[the remark this answers is not in the document]
Comment (ouchadam) [reply vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt]: updated :+1: 2c843985cae65993604351aaa0215d0675992724

**Pull request merged**